core/blockchain_repair_test.go

```diff
@@ -544,17 +563,29 @@ func testRepair(t *testing.T, tt *rewindTest, snapshots bool) {
 	// If sidechain blocks are needed, make a light chain and import it
 	var sideblocks types.Blocks
 	if tt.sidechainBlocks > 0 {
-		sideblocks, _, _ = GenerateChain(gspec.Config, gspec.ToBlock(), engine, rawdb.NewMemoryDatabase(), tt.sidechainBlocks, 10, func(i int, b *BlockGen) {
+		genDb := rawdb.NewMemoryDatabase()
+		gspec.MustCommit(genDb, trie.NewDatabase(genDb, nil))
+		sideblocks, _, err = GenerateChain(gspec.Config, gspec.ToBlock(), engine, genDb, tt.sidechainBlocks, 10, func(i int, b *BlockGen) {
 			b.SetCoinbase(common.Address{0x01})
+			tx, err := types.SignTx(types.NewTransaction(b.TxNonce(addr1), common.Address{0x01}, big.NewInt(10000), params.TxGas, common.Big1, nil), signer, key1)
+			require.NoError(err)
+			b.AddTx(tx)
```

Comment:
Had to make these changes since in geth, the roots are different because of the coinbase payment.
Here we just keep on with the empty trie and that doesn't work for the pathdb (doesn't expect repeated roots)